src/joinboost/app.py

```diff
@@ -124,6 +125,20 @@ def compute_rmse(self, test_table: str):
         predict = self.cjt.exe.execute_spja_query(predict_agg, [view], mode=4)
         return self.cjt.exe.execute_spja_query(from_tables=[predict], 
                                                mode=3)[0]
+    
+    # mode = 1 takes the full join as input
+    # mode = 2 takes the join graph as input (assume fact table)
+    # mode = 3 takes the fact table as input (and automatically join it with dimensional tables used in training)
+    def predict(self, data, mode: int = 1):
```

Comment:
Things could get complex for the API.
Firstly, what's the type of input? It could be [fullJoin, JoinGraph, FactTable].

Secondly, what's the format of the input? 
Let's assume that
For [fullJoin, FactTable], let it be the table name. For [JoinGraph], let it be a join graph class.
(And check if the format is correct)

Thirdly, what's the output format? Shall we store them in the database, or return the table as pandas?
Let's have a option to return [tablename, numpy array]

Reply:
Maybe the signature is like 
def predict(self, data, input_mode, output_mode):

Reply:
For the 3rd point, I follow the lightGBM's API https://github.com/microsoft/LightGBM/blob/4f47547c88cc72de4b8f04de8de4e2b79a9240e7/python-package/lightgbm/basic.py#L812, which returns a numpy array. I feel like we should not store anything into the database for the predict function.